docs/glossary.rst

```diff
@@ -77,6 +77,22 @@ Astropy Glossary
       :class:`~astropy.time.Time`, e.g. `str`, array-like[str], `~datetime.datetime`, or
       `~numpy.datetime64`.
 
+   trait type
+      In short, a trait type is an orthogonal mixin class. Strictly following the
+      definition in :mod:`mypyc`, a trait type is not a Python type, but a C type.
```

Comment:
Sphinx can't figure out the links to Mypyc. Maybe it's best not to mention Mypyc here at all. The glossary entry should be more focused on traits that are not decorated with `@trait`. The glossary entry not really needed for the decorated traits because it would be simple enough to refer to the decorator documentation.

Reply:
Done.